core/lib/Thelia/Core/Template/TemplateHelper.php

```diff
@@ -12,321 +12,94 @@
 
 namespace Thelia\Core\Template;
 
-use Symfony\Component\Filesystem\Filesystem;
-use Thelia\Model\ConfigQuery;
-use Thelia\Log\Tlog;
-use Thelia\Core\Translation\Translator;
-
-class TemplateHelper
+/**
+ * Class TemplateHelper
+ * @author Franck Allimant <franck@cqfdev.fr>
+ * @package Thelia\Core\Template
+ *
+ * @deprecated use TheliaTemplateHelper service (thelia.template_helper) instead. This class will be DELETED in 2.3
+ */
+class TemplateHelper implements TemplateHelperInterface
 {
-    const WALK_MODE_PHP = 'php';
-    const WALK_MODE_TEMPLATE = 'tpl';
-
     /**
      * This is a singleton
-
      */
     private static $instance = null;
 
-    private function __construct()
+    private function __construct(TemplateHelperInterface $templateHelper)
     {
+        $this->templateHelper = $templateHelper;
     }
 
+    /**
+     * @deprecated use TheliaTemplateHelper service (thelia.template_helper) instead.
+     */
     public static function getInstance()
     {
         if (self::$instance == null) {
-            self::$instance = new TemplateHelper();
+            throw new \LogicException("This class should be initialized before getInstance() call.");
```

Comment:
If you don't want BC break, why did you throw this exception ? I think you can just tag this method as deprecated.

If you want you can trigger a notice error like here : https://github.com/thelia/thelia/blob/master/core/lib/Thelia/Controller/BaseController.php#L565